WordPress/gutenberg
# Site Editor: Add Dropdown to Create Generic Templates

## Description

Part of #26270

Add a very basic interface to create generic templates from the Site Editor navigation sidebar.
Clicking on the `+` button will open a dropdown listing all _missing_ [generic templates](https://github.com/WordPress/gutenberg/blob/e2d8b7a2c544d1142e7bfbbb1422a609d78e050d/packages/edit-site/src/utils/get-template-info/constants.js), filling them with the hierarchically closest template content, plus a "General" template that creates a blank template.

cc @jameskoster (for some reasons I can't add you to the reviewers)

### Major Notes

1. New templates are now `draft`. The idea is to avoid having them immediately applied to the site, while they might still be WIP.
Though, we don't have a proper way to work with draft templates (e.g. the "save" button is only "update", there is no publish flow).
2. We'll probably eventually need to have "fixed" slugs for `$custom` templates, otherwise there might be the risk that the user accidentally creates one with a "reserved" name (e.g. "Front Page" which would be slugified into `front-page`).

## How has this been tested?

- Enable the Full Site Editing experiment.
- Open the Site Editor.
- Click on the WP/site icon button, and then on "Templates".
- Click on the `+` button near the Templates menu title.
- Make sure a dropdown with a list of generic templates shows up.
- Make sure the templates already available on the site are not listed.
- Click on one of the "named" templates.
- Make sure the editor content is replaced with the appropriate template's content (e.g. `archive` should insert the `index` template content), and the document title shows the new template title.
- Click on the "General" template.
- Make sure the editor content is empty, and the document title is "New Template".

## Screenshots 

<img width="546" alt="Screenshot 2020-10-23 at 12 50 26" src="https://user-images.githubusercontent.com/2070010/97000225-5ee9ed80-152e-11eb-9457-a3ce5e41a263.png">

## Types of changes





## Checklist:
